### ISSUE TITLE
Add JetBrains IDE button to all starter templates

### ISSUE THREAD

**Issue by chrifro** [coder/coder#14822]
**Issue**:
users might be surprised to not see an option to use a JetBrains IDE
![image](https://github.com/user-attachments/assets/492f8379-082e-4bc5-b64f-85c002371c48)

**Possible solution:**

<img width="338" alt="Screenshot 2024-09-26 at 14 23 25" src="https://github.com/user-attachments/assets/cc8bbef5-f05e-47e0-a3e0-0739318a613c">

Note: can we give any feedback to the user in case an IDE is not installed yet?

**Comment by ammario** [comment]
Hm, a part of the reason we don't do this is it has been a bit repetitive to add apps to every template. Perhaps we can contain a reasonable set of default apps in a model and reference that module from each of our example templates.

**Comment by matifali** [comment]
@ammario, What do you think about just using the [module](https://registry.coder.com/modules/jetbrains-gateway) from registry.coder.com/modules?

We do not have modules in all of our example templates but they do exist in some.

**Comment by ammario** [comment]
Do you mean copying in that module declaration into all of our templates?

**Comment by matifali** [comment]
@ammario No. I mean directly using the modules hosted on registry. like,

```tf
module "jetbrains_gateway" {
  source         = "registry.coder.com/modules/jetbrains-gateway/coder"
  version        = "1.0.13"
  agent_id       = coder_agent.example.id
  agent_name     = "example"
  folder         = "/home/coder/example"
  jetbrains_ides = ["CL", "GO", "IU", "PY", "WS"]
  default        = "GO"
}
```

**Comment by ammario** [comment]
yeah.. our example templates shouldn't be full of duplicated content.. that's not the kind of example we want to set for our users.
